solr/solrj/src/java/org/apache/solr/common/util/ValidatingJsonMap.java

```diff
@@ -348,10 +348,12 @@ public boolean equals(Object that) {
     return that instanceof Map && this.delegate.equals(that);
   }
 
-//  @Override
-//  public int hashCode() {
-//    throw new UnsupportedOperationException("TODO unimplemented ValidatingJsonMap.hashCode");
-//  }
+  //TODO: Really uncertain about this. Hashing the map itself seems
+  // about as expensive as resolving with equals.
```

Comment:
`hashCode()` and `equals()` are not used for the same type of computations, it's not one _or_ the other, so their relative costs are not necessarily an implementation criteria.

Reply:
Right. Still, every time a hashcode is computed, it iterates the entire map which could be arbitrarily large. I don't see much other choice though.